recipe/meta.yaml

```
@@ -1,51 +1,97 @@
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/i/ipyleaflet/ipyleaflet-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  - url: https://pypi.org/packages/source/i/ipyleaflet/ipyleaflet-{{ version }}.tar.gz
+    sha256: b3b83fe3460e742964c2a5924ea7934365a3749bb75310ce388d45fd751372d2
+    folder: ipyleaflet
+  - url: https://pypi.org/packages/source/j/jupyter_leaflet/jupyter_leaflet-{{ version }}.tar.gz
+    sha256: b09b5ba48b1488cb61da37a6f558347269eb53ff6d64dc1a73e005ffc4420063
+    folder: jupyter_leaflet
+
 
 build:
   number: 0
-  skip: true # [py<37 or s390x]
-  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+  skip: true # [py<38 or s390x]
 
 requirements:
   host:
     - python
     - pip
+    - hatchling
     - setuptools
     - wheel
     - jupyter-packaging >=0.12,<1
-  run:
-    - python
-    - ipywidgets >=7.6.0,<9
-    - traittypes >=0.2.1,<0.3.0
-    - xyzservices >=2021.8.1
-    - branca >=0.5.0
-
-test:
-  imports:
-    - ipyleaflet
-  requires:
-    - pip
-    - m2-grep # [win]
-    - jupyterlab
-  commands:
-    - pip check
-    - jupyter labextension list
-    - jupyter labextension list 2>&1 | grep -ie " jupyter-leaflet.*OK"
 
+outputs:
+  - name: jupyter_leaflet
+    version: {{ version }}
+    build:
+      script: cd jupyter_leaflet && python -m pip install . --no-deps --no-build-isolation -vv
+    requirements:
+      host:
+        - python
+        - pip
+        - nodejs >=18,<19
+        - hatch-jupyter-builder >=0.8.1
+        - hatchling
+        - jupyterlab >=4,<5
+        - hatch-nodejs-version >=0.3.2
+    test:
+      imports:
+        - jupyter_leaflet
+      requires:
+        - python
+        - pip
+      commands:
+        - pip check
+        - test -f ${PREFIX}/share/jupyter/nbextensions/jupyter-leaflet/extension.js                             # [unix]
+        - test -f ${PREFIX}/share/jupyter/nbextensions/jupyter-leaflet/index.js                                 # [unix]
+        - test -f ${PREFIX}/share/jupyter/labextensions/jupyter-leaflet/package.json                            # [unix]
+        - if not exist %PREFIX%\\share\\jupyter\\nbextensions\\jupyter-leaflet\\extension.js (exit 1)           # [win]
+        - if not exist %PREFIX%\\share\\jupyter\\nbextensions\\jupyter-leaflet\\index.js (exit 1)               # [win]
+        - if not exist %PREFIX%\\share\\jupyter\\labextensions\\jupyter-leaflet\\package.json (exit 1)          # [win]
+
+  - name: ipyleaflet
+    version: {{ version }}
+    build:
+      script: cd ipyleaflet && python -m pip install . --no-deps --no-build-isolation -vv
+    requirements:
+      host:
+        - python
+        - pip
+        - hatchling
+      run:
+        - python
+        - ipywidgets >=7.6.0,<9
+        - traittypes >=0.2.1,<0.3.0
+        - xyzservices >=2021.8.1
+        - branca >=0.5.0
+        - jupyter_leaflet
+      run_constrained:
+        - jupyter_leaflet ={{ version }}
```

```suggestion
        - {{ pin_subpackage('jupyter_leaflet', max_pin='x.x') }}
```
This should yield expected `>=0.19,<0.20`